course5/homework6/mystr.cc

```diff
@@ -0,0 +1,104 @@
+#include "mystr.h"
+#include <memory>
+#include <cmath> 
+#include <cstring>
+
+using namespace mystring;
+
+MyString::MyString()
+{
+	mStr = std::make_unique<char[]>(1); //length 1 for zero terminator
+	mStr[0] = '\0';
+}
+
+// Function to create a new unique pointer, memset it and initialize it with a 
+// new string. 
+// Repeatedly needed for copy/move assignment and construction
+void MyString::Reset_mStr(int len, const char* str)
+{
+	mStr = std::make_unique<char[]>(len+1); // add 1 for \0 termination
+	std::memset(mStr.get(), 0, len + 1);
+	std::strncpy(mStr.get(), str, len);
+}
+
+MyString::MyString(const MyString& str)  // copy constructor
+{
+	Reset_mStr(str.length(), str.c_str());
+}
+
+MyString& MyString::operator = (const MyString& str)  // copy assignment
+{
+	Reset_mStr(str.length(), str.c_str());
+	return *this;
+}
+	
+
+MyString:: MyString(const MyString&& str)  // move constructor
```

Comment:
This is not move constructor. Move constructor takes non-const rvalue referenvce (```MyString&&```), so that you can modify the source object.
You need to be able to modify the source object to steal from it.
You implementation simply copies from the source, leaving the source intact; it is effectively a copy constructor.

The idea behind move constructor is to avoid O(N) copy, and replace it with O(1) stealing.